File: python/unclean.py

from enum import Enum
from random import random
import math
import time



Shape = Enum('Shape', ['SQUARE', 'RECTANGLE', 'TRIANGLE','CIRCLE'])


def area(shape):
    if shape["type"] == Shape.SQUARE:
            return shape["width"]*shape["width"]
    elif shape["type"] == Shape.RECTANGLE:
            return  shape["width"]*shape["height"]
    elif shape["type"] == Shape.TRIANGLE:
            return 0.5* shape["width"]*shape["height"]
    elif shape["type"] == Shape.CIRCLE:
            return math.pi* shape["width"]*shape["width"]
    

shapes=[]

for i in range(1000000):
    shapes.append({"type":Shape.SQUARE,"width":random()})
    shapes.append({"type":Shape.RECTANGLE,"width":random(),"height":random()})
    shapes.append({"type":Shape.TRIANGLE,"width":random(),"height":random()})
    shapes.append({"type":Shape.CIRCLE,"width":random()})


start = time.time()
totalArea = 0;
for shape in shapes:
    totalArea = totalArea + area(shape)
end = time.time()

print("Processing time in ms =", (end - start)*1000, " ms");

File: js/unclean.js

const SQUARE = 0;
const RECTANGLE = 1;
const TRIANGLE = 2;
const CIRCLE = 3;

class Shape {
  constructor(type, width, height) {
    this.type = type;
    this.width = width;
    this.height = height;
  }
  width = 0;
  height = 0;
  type = SQUARE;
}

function area(shape) {
  switch (shape.type) {
    case SQUARE:
      return shape.width * shape.width;
    case RECTANGLE:
      return shape.width * shape.height;
    case TRIANGLE:
      return 0.5 * shape.width * shape.height;
    case CIRCLE:
      return Math.PI * shape.width * shape.width;
  }
}

function areaNotClean() {

  const shapes = new Array();
  for (let i = 0; i < 1000000; i++) {
    shapes.push(new Shape(SQUARE, Math.random(), 0));
    shapes.push(new Shape(RECTANGLE, Math.random(), Math.random()));
    shapes.push(new Shape(TRIANGLE, Math.random(), Math.random()));
    shapes.push(new Shape(CIRCLE, Math.random(), 0));
  }
  let totalArea = 0;
  const start = new Date().valueOf();
  for (let i = 0; i < shapes.length; i++) {
    totalArea += area(shapes[i]);
  }
  const end = new Date().valueOf();

  console.log("Processing time in ms =", end - start, " ms");
}
areaNotClean();
